lib/jackalope/session.ex

```diff
@@ -268,11 +268,12 @@ defmodule Jackalope.Session do
 
   defp expired?(expiration), do: Expiration.after?(Expiration.expiration(0), expiration)
 
-  defp retry_error?({{:subscribe, _}, _}, reason) do
-    # reason is a list of {status, {topic, qos}}
-    # if all the statuses are :ok or :access_denied, we don't want to retry
-    Enum.any?(reason, fn {status, _topic} -> status not in [:ok, :access_denied] end)
+  # When reasons is a list of {status, {topic, qos}},
```

Comment:
The error responses here are a little strange for typical Elixir functions. Are these handling the return value of one Tortoise function?

Reply:
Tortoise311 is a little strange. That's how it sends subscription errors to Jackalope's TortoiseClient, as tuple messages. ToirtoiseClient forwards them to Session for reporting. The error component from Tortoise311 is either a list of reasons or a :timeout error. The :timeout case was not being handled.